apollo-router-core/src/services/http_compat.rs

```diff
@@ -1,78 +1,25 @@
-//! wrapper typpes for Request and Response from the http crate to improve their usability
+//! Wrapper types for [`http::Request`] and [`http::Response`] from the http crate.
+//!
+//! To improve their usability.
 
 use std::{
     cmp::PartialEq,
     hash::Hash,
     ops::{Deref, DerefMut},
-    str::FromStr,
 };
 
-use http::{
-    header::HeaderName, request::Builder, uri::InvalidUri, HeaderMap, HeaderValue, Method, Uri,
-    Version,
-};
+use http::{header::HeaderName, request::Builder, uri::InvalidUri, HeaderValue, Version};
 
 #[derive(Debug)]
 pub struct Request<T> {
     // The goal of having a copy of the url is to keep the right type for `ReqwestSubgraphService` and avoid re-parsing.
     // This url will stay the same than the uri in inner because we only can set a new url with `set_url` method
-    pub(super) url: Uri,
-    inner: http::Request<T>,
+    // pub(super) url: Uri,
```

Comment:
I think as we are not using `reqwest` anymore, we can get rid of this. Only if we keep `inner` private to ensure we have a correct request